drivers/filter/admv8818/admv8818.c

```diff
@@ -0,0 +1,423 @@
+					ADMV8818_HPF_WR0_MSK,
+					no_os_field_prep(ADMV8818_HPF_WR0_MSK, hpf_step));
+}
+
+/**
+ * @brief Get the HPF Frequency.
+ * @param dev - The device structure.
+ * @param freq - The HPF Frequency.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int admv8818_read_hpf_freq(struct admv8818_dev *dev, unsigned long long *freq)
+{
+	unsigned int hpf_band, hpf_state;
+	uint8_t data;
+	int ret;
+
+	ret = admv8818_spi_read(dev, ADMV8818_REG_WR0_SW, &data);
+	if (ret)
+		return ret;
+
+	hpf_band = no_os_field_get(ADMV8818_SW_IN_WR0_MSK, data);
+	if (!hpf_band) {
+		*freq = 0;
+		return ret;
+	}
+
+	ret = admv8818_spi_read(dev, ADMV8818_REG_WR0_FILTER, &data);
+	if (ret)
+		return ret;
+
+	hpf_state = no_os_field_get(ADMV8818_HPF_WR0_MSK, data);
+
+	*freq = (freq_range_hpf[hpf_band - 1][1] - freq_range_hpf[hpf_band - 1][0]) /
+		15;
+	*freq = freq_range_hpf[hpf_band - 1][0] + (*freq * hpf_state);
+
+	return 0;
+}
+
+/**
+ * @brief Set the LPF Frequency.
+ * @param dev - The device structure.
+ * @param freq - The LPF Frequency to be set.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int admv8818_lpf_select(struct admv8818_dev *dev, unsigned long long freq)
+{
+	unsigned int lpf_step = 0, lpf_band = 0, i, j;
+	unsigned long long freq_step;
+	int ret;
+
+	if (freq > freq_range_lpf[3][1])
+		goto lpf_write;
+
+	if (freq < freq_range_lpf[0][0]) {
+		lpf_band = 1;
+
+		goto lpf_write;
+	}
+
+	for (i = 0; i < 4; i++) {
+		if (freq > freq_range_lpf[i][0] && freq < freq_range_lpf[i][1]) {
+			lpf_band = i + 1;
+			freq_step = (freq_range_lpf[i][1] - freq_range_lpf[i][0]) / 15;
+
+			for (j = 0; j <= 15; j++) {
+				if (freq < (freq_range_lpf[i][0] + (freq_step * j))) {
+					lpf_step = j;
+					break;
+				}
+			}
+			break;
+		}
+	}
+
+lpf_write:
+	ret = admv8818_spi_update_bits(dev, ADMV8818_REG_WR0_SW,
+				       ADMV8818_SW_OUT_SET_WR0_MSK |
+				       ADMV8818_SW_OUT_WR0_MSK,
+				       no_os_field_prep(ADMV8818_SW_OUT_SET_WR0_MSK, 1) |
+				       no_os_field_prep(ADMV8818_SW_OUT_WR0_MSK, lpf_band));
+	if (ret)
+		return ret;
+
+	return admv8818_spi_update_bits(dev, ADMV8818_REG_WR0_FILTER,
+					ADMV8818_LPF_WR0_MSK,
+					no_os_field_prep(ADMV8818_LPF_WR0_MSK, lpf_step));
+}
+
+/**
+ * @brief Get the HPF Frequency.
+ * @param dev - The device structure.
+ * @param freq - The HPF Frequency.
+ * @return Returns 0 in case of success or negative error code.
+ */
+int admv8818_read_lpf_freq(struct admv8818_dev *dev, unsigned long long *freq)
+{
+	unsigned int lpf_band, lpf_state;
+	uint8_t data;
+	int ret;
+
+	ret = admv8818_spi_read(dev, ADMV8818_REG_WR0_SW, &data);
+	if (ret)
+		return ret;
+
+	lpf_band = no_os_field_get(ADMV8818_SW_OUT_WR0_MSK, data);
+	if (!lpf_band) {
+		*freq= 0;
+		return ret;
+	}
+
+	ret = admv8818_spi_read(dev, ADMV8818_REG_WR0_FILTER, &data);
+	if (ret)
+		return ret;
+
+	lpf_state = no_os_field_get(ADMV8818_LPF_WR0_MSK, data);
+
```

Comment:
nitpick: would it be useful to add a comment here which explains these operations?